modules/codefuelBidAdapter.js

```diff
@@ -0,0 +1,182 @@
+import * as utils from '../src/utils.js';
+// import { config } from '../src/config.js';
+import {registerBidder} from '../src/adapters/bidderFactory.js';
+import { BANNER } from '../src/mediaTypes.js';
+const BIDDER_CODE = 'codefuel';
+const CURRENCY = 'USD';
+
+export const spec = {
+  code: BIDDER_CODE,
+  supportedMediaTypes: [ BANNER ],
+  aliases: ['ex'], // short code
+  /**
+         * Determines whether or not the given bid request is valid.
+         *
+         * @param {BidRequest} bid The bid params to validate.
+         * @return boolean True if this is a valid bid, and false otherwise.
+         */
+  isBidRequestValid: function(bid) {
+    if (bid.nativeParams) {
+      return false;
+    }
+    return !!(bid.params.placementId || (bid.params.member && bid.params.invCode));
+  },
+  /**
+         * Make a server request from the list of BidRequests.
+         *
+         * @param {validBidRequests[]} - an array of bids
+         * @return ServerRequest Info describing the request to the server.
+         */
+  buildRequests: function(validBidRequests, bidderRequest) {
+    const page = bidderRequest.refererInfo.referer;
+    // const page = 'https://www.smartreadz.com/best-and-worst-foods-for-diabetes/'
+    const domain = getDomainFromURL(page)
+    const ua = navigator.userAgent;
+    const devicetype = getDeviceType()
+    const publisher = setOnAny(validBidRequests, 'params.publisher');
+    const cur = CURRENCY;
+    // const endpointUrl = 'http://localhost:5000/prebid'
+    const endpointUrl = 'https://ai-i-codefuel-ds-rtb-us-east-1-k8s-internal.seccint.com/prebid'
+    const timeout = bidderRequest.timeout;
+
+    const imps = validBidRequests.map((bid, id) => {
+      bid.netRevenue = 'net';
```

Comment:
I wouldn't recommend updating the **validBidRequests** obj inside the map function

Reply:
I don't see where validBidRequests is updated, please clarify

Reply:
We run throuh the `validBidRequests` array and on each iteration we mutate the `netRevenue` prop. 
`bid.netRevenue = 'net';` (bid is a reference for `validBidRequests` array entry).
When using a map functions It's a better practice to use the returned object (a deep copy) and not mutating the original.
If we do want to mutate the `validBidRequests` array, I'd run a separte forEach loop

Reply:
I see, please fix it.